libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/Preds/PParams.hs

```diff
@@ -79,9 +79,9 @@ genPParams proof tx bb bh = do
           , Prices (Script.Prices (nonNegativeInterval 1.0) (nonNegativeInterval 1.0))
           , defaultCostModels proof
           , MaxValSize 1000
-          , MaxTxSize tx
-          , MaxBBSize bb
-          , MaxBHSize bh
+          , MaxTxSize (fromIntegral tx) -- In the Model these are Natural
+          , MaxBBSize (fromIntegral bb) -- But in the PParams, they are
+          , MaxBHSize (fromIntegral bh) -- Word32, Word32, and Word16
```

Comment:
So I did this
`fromIntegralBounded :: forall a b.
  (HasCallStack, Integral a, Show a, Integral b, Bounded b, Show b) =>
  String -> a -> b
fromIntegralBounded name x
  | toInteger (minBound :: b) <= xi && xi <= toInteger (maxBound :: b) = fromIntegral x
  | otherwise = 
    error $ "While converting "++name++", "++show x <> " is out of bounds: " <> show (minBound :: b, maxBound ::b)
  where
    xi = toInteger x`

This way the error tells what you were converting.